crates/p2p/src/swarm/mod.rs

```diff
@@ -473,7 +500,7 @@ where
             &msg, /* do net set timeout for local peer */ false,
         )
         .await
-        .context(RepositorySnafu)?;
+        .map_err(Error::Repository)?;
```

Comment:
If you added `#[from]` for `RepositoryError`, just `?` wihtou `map_err` should work fine